resolver/wasm/wasm.go

```diff
@@ -80,6 +80,8 @@ func (r *Resolver) Close() {
 // Eval performs an evaluation using the provided input and the Wasm module
 // associated with this Resolver instance.
 func (r *Resolver) Eval(ctx context.Context, input resolver.Input) (resolver.Result, error) {
+	input.Metrics.Timer("wasm_resolver_eval").Start()
+	defer input.Metrics.Timer("wasm_resolver_eval").Stop()
```

Comment:
This metric seems a bit redundant. I think we're essentially measuring the time to convert the AST input value into a native go value. Along those lines, I'm tempted to just pass AST input value into the SDK for two reasons:

- We'll need to do this anyway to support sets in the input
- We'll avoid the need to roundtrip the input value (in theory we could serialize from AST input value directly into the Wasm address space--in the short term we can just call `String()` and copy the bytes.)

Reply:
> This metric seems a bit redundant

+1, It was a remnant of the first pass at plumbing these through to only the resolver, but after pushing them all the way down into the wasm sdk stuff it doesn't make sense anymore.

> I'm tempted to just pass AST input value into the SDK for two reasons

Sounds good to me, making the changes now.

Reply:
Latest commit includes changes to support rego types in and out of the wasm evaluation. We're using the input stringer and copying bytes on the way in, then `opa_value_dump` and `ast.ParseTerm` so we can get set values on the way out.